### ISSUE TITLE
Having an issue with app being recognized as Health App [Android]

### ISSUE THREAD

**Issue by christian-hess-94** [transistorsoft/react-native-background-geolocation#2031]
## Your Environment
* Plugin version: ^4.16.0
* Platform: Android
* OS version: 14
* Device manufacturer / model: Google Pixel 5
* React Native version (`react-native -v`): 0.73.8
* Plugin config
```javascript <!-- Syntax-highlighting:  paste your code below -->
{
  // Geolocation Config
  desiredAccuracy: BackgroundGeolocation.DESIRED_ACCURACY_HIGH,
  distanceFilter: 100,
  useSignificantChangesOnly: true,
  disableElasticity: false,
  // Activity Recognition
  stopTimeout: 5,
  disableMotionActivityUpdates: true,

  // Application config
  debug: false, // <-- enable this hear sounds for background-geolocation life-cycle.
  logLevel: BackgroundGeolocation.LOG_LEVEL_VERBOSE,
  stopOnTerminate: false, // <-- Allow the background-service to continue tracking when user closes the app.
  startOnBoot: false, // <-- Auto start tracking when device is powered-up.
  url: `${API_URL}/providers/${userSession?.id}/location`,
  batchSync: true, // <-- [Default: false] Set true to sync locations to server in a single HTTP request.
  maxBatchSize: 5,
  autoSync: true, // <-- [Default: true] Set true to sync each location to server as it arrives.
  headers: {
    // <-- Optional HTTP headers
    'X-TUP-Auth-Version': '1',
    Authorization: `Bearer ${userSession?.access_token}`,
    Accept: 'application/json',
    'Content-Type': 'application/json',
    'TUP-Device-Os': Platform.OS, // android || ios
    'TUP-Device-Id': getDeviceId(),
    'TUP-Build-Number': getBuildNumber(),
    'TUP-App-Version': getVersion(),
  },
  params: {},
```

## Expected Behavior

We want to be able to deploy our app into production like we did before with our CI/CD workflow

## Actual Behavior

Our issue is not with the plugin itself working in the app, but when trying to deploy it to the play store.
We have recently update our app to react-native version 0.73.8 and we are getting the following error upon trying to deploy the app to production
![image](https://github.com/transistorsoft/react-native-background-geolocation/assets/34290680/2c31f37f-4788-4a0d-aa27-b0a633722393)

## Steps to Reproduce

There is no way to reproduce this unless when trying to deploy to Play Store

## Context

We recently updated our app to use react-native version 0.73.8. The app is working fine and the plugin too, however we get an error when trying to run it via our CICD pipeline. The app is built fine, however when trying to publish it to play store, we get the following error:
![image](https://github.com/transistorsoft/react-native-background-geolocation/assets/34290680/c0e4af5c-8e00-4b95-8d6a-75ec5de32f4b)

Upon doing additional checks, we noticed that PlayStore is recognizing our app as a Health App due to it's usage of the android.permission.ACTIVITY_RECOGNITION permission. We have checked our code and we do not have this permission anywhere in our Manifest file, although it is configured in the plugin. We also have disabled the motionActivityUpdates as per the config file.

Is there a way for us to remove this permission so our app is not recognized a health app anymore?

This is what shows up in our play console:
![image](https://github.com/transistorsoft/react-native-background-geolocation/assets/34290680/06550da7-4aa9-47a2-b51b-9613077f6eea)

## Debug logs

<details><summary>Logs</summary>

![image](https://github.com/transistorsoft/react-native-background-geolocation/assets/34290680/ba6686f8-2c0e-494f-b6a5-700468319fc6)

</details>

**Comment by christocracy** [comment]
The plug-in uses requires this permission to use Motion API for automatically toggling location-services ON/OFF.

**Comment by christian-hess-94** [comment]
so should we just set the app as a Health app and fill the forms?

**Comment by christocracy** [comment]
Yes.

**Comment by aaldavea** [comment]
I have the same problem, play store is detecting my application as if it were HEALTH, however I require the ACTIVITY_RECOGNITION permission since the react-native-background-geolocation library needs it.

It will be enough if I use this description "The plug-in uses requires this permission to use Motion API for automatically toggling location-services ON/OFF." to justify why I'm using the ACTIVITY_RECOGNITION permission: ? since I need to publish the application in the Play store.

<img width="1017" alt="healh" src="https://github.com/transistorsoft/react-native-background-geolocation/assets/92137801/17113fb2-f91f-4db0-9a65-d2a94a74e5d9">

**Comment by christocracy** [comment]
I wouldn’t use the phrase “the plug-in”.  Reviewers don’t know or care about plug-ins.  It’s “my app”.

**Comment by aaldavea** [comment]
I'm going to use the following justification: "My app requires the ACTIVITY_RECOGNITION permission to use the Motion API to automatically turn location services on/off based on user activity. This optimizes battery life and improves tracking accuracy. We'll use this permission responsibly." I think it is important how it should be justified since it is a problem that those who use the library are having.

Does the justification seem good to you?

**Comment by christocracy** [comment]
It sounds perfect.  I would also add a shorter phrase about it on the  “prominent disclosure”.

see step 5 at my blog here:  
https://transistorsoft.medium.com/new-google-play-console-guidelines-for-sensitive-app-permissions-d9d2f4911353

**Comment by aaldavea** [comment]
It was approved

**Comment by christocracy** [comment]
Good news:  I've found a way to enable motion-activity updates in the background without `FOREGROUND_SERVICE_HEALTH` permission.  Will be released in the next version `v4.16.3`.

**Comment by christocracy** [comment]
See my latest [blog-post about this](https://transistorsoft.medium.com/google-play-store-required-declarations-for-foreground-services-and-health-97da6e2abd5a)

**Comment by DanW82** [comment]
fyi, if your having issues completing the google health app submission then it needs to be complete by the ACCOUNT OWNER.
The UI doesn't tell you that but if you have issues that might be the the cause.

I can normally complete these declaration forms but this one is different. I could fill in the fields on the form but the save button is disabled. Tried different browsers, other people tried but in the end only the account owner could do it.

**Comment by mwmcode** [comment]
> See my latest [blog-post about this](https://transistorsoft.medium.com/google-play-store-required-declarations-for-foreground-services-and-health-97da6e2abd5a)

Hi Chris, I see Google Play is again asking for activity recognition declaration even though I have set it to be removed (`<uses-permission android:name="android.permission.ACTIVITY_RECOGNITION" tools:node="remove" />`) for a few months now. 
Any idea how to debug this?

<img width="898" alt="Screenshot 2024-09-12 at 10 59 47 AM" src="https://github.com/user-attachments/assets/2c6d036d-19fa-4511-8a66-893643625527">

`android/app/src/main/AndroidManifest.xml`

<img width="918" alt="Screenshot 2024-09-12 at 10 53 23 AM" src="https://github.com/user-attachments/assets/83ec508e-39f0-47f6-a78f-a398f7563e11">

**Comment by christocracy** [comment]
I just tested this on my demo app, adding the following to my `AndroidManifest`:

```diff
<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools">

+    <uses-permission android:name="android.permission.ACTIVITY_RECOGNITION" tools:node="remove" />
</manifest>
```

Next, I install a tool to analyze the my `app-release.apk`

```console
$ wget https://github.com/google/android-classyshark/releases/download/8.2/ClassyShark.jar
$ java -jar ClassyShark.jar -open <app-release.apk>
```

I see that `<uses-permission android:name="android.permission.ACTIVITY_RECOGNITION" />` does not exist in my `AndroidManifest.xml`.  For more information on using `tools:node="remove"`, see the [Android docs: "Manage Manifest Files"](https://developer.android.com/build/manage-manifests#node_markers)

<img width="1515" alt="Screenshot 2024-09-12 at 11 31 42 AM" src="https://github.com/user-attachments/assets/b71fbac6-73b3-4375-b0be-ff20efc054ac">

**Comment by christocracy** [comment]
You do not need to worry about those permissions.  One is only used by Huawei devices running the HMS operating system and the other is for older versions of Android.

Your warning message from Play Store specifically complains about a permission named `android.permission.ACTIVITY_RECOGNITION`.  Those two permissions are not that.

**Comment by mwmcode** [comment]
Resolved. Appreciate your quick response 🙏 

If anyone comes across this here, the issue was due to _another release track_ hand an older version with that permission. The 'solution' was to release a new version on that track too, so the request for declaration goes away.

**Comment by LaurentBonnot** [comment]
Hi @christocracy 

This week I did an update to my app. It's been more than 2 years the app is public. And the app released was rejected. Google complain now about the ACTIVITY_RECOGNITION. 

![image](https://github.com/user-attachments/assets/286ad66d-0f6b-4921-b94c-133d1f880ead)


I changed the app content declaration following your blog post
![image](https://github.com/user-attachments/assets/f66bf86b-6a4b-4b3a-bd1d-d545af546adc)

But even with that Google constantly reject my submission. It seems to be automated now, because when I submit, it takes a couple of minutes and I have their response (compare to multiple days before)

**Comment by christocracy** [comment]
From Android's own API docs:  [Detect when users start or end an activity](https://developer.android.com/develop/sensors-and-location/location/transitions)

<img width="953" alt="Screenshot 2024-09-16 at 11 33 09 AM" src="https://github.com/user-attachments/assets/38642bca-639e-46b7-9af5-f569c800a178">

**Comment by chrisbiscuit** [comment]
Hi all,

I wanted to share a helpful post related to a Google Play rejection issue I faced, which could assist others dealing with similar challenges. After struggling with repeated rejections over the Health Declaration form, I was able to resolve it by removing the **"Activity & Fitness"** category and clarifying that our app doesn’t provide any health features.

I’ve detailed the steps I took in the following GitHub post, which may be useful for anyone facing similar issues with Google Play’s policies:

[How we resolved Google Play rejection](https://github.com/transistorsoft/capacitor-background-geolocation/issues/289)

Hope it helps!
